src/hotspot/share/opto/memnode.hpp

```diff
@@ -1388,6 +1401,32 @@ class InitializeNode: public MemBarNode {
                                PhaseGVN* phase);
 
   intptr_t find_next_fullword_store(uint i, PhaseGVN* phase);
+
+  // Iterate with i over all NarrowMemProj uses calling callback
+  template <class Callback, class Iterator> NarrowMemProjNode* apply_to_narrow_mem_projs_any_iterator(Iterator i, Callback callback) const {
+    auto filter = [&](ProjNode* proj) {
+      if (proj->is_NarrowMemProj() && callback(proj->as_NarrowMemProj()) == BREAK_AND_RETURN_CURRENT_PROJ) {
+        return BREAK_AND_RETURN_CURRENT_PROJ;
+      }
+      return CONTINUE;
+    };
+    ProjNode* res = apply_to_projs_any_iterator(i, filter);
+    if (res == nullptr) {
+      return nullptr;
+    }
+    return res->as_NarrowMemProj();
+  }
+
+public:
+
+  template <class Callback> void for_each_narrow_mem_proj_with_new_uses(Callback callback) const {
```

Comment:
Can you add a code comment what the "with new uses" part means? Probably that if we add more uses during iteration, we will eventually also visit those?